R/indexing.R

```diff
@@ -322,3 +357,303 @@ BasicIndexer <- R6::R6Class("BasicIndexer",
+                              #'   An iterator over the dimensions of an array
+                              #' @return A list of ChunkProjection objects
+                              iter = function() {
+                                
+                                # TODO: use generator/yield features from async package
+                                result <- list()
+                                
+                                # dim_indexers is a list of DimIndexer objects.
+                                # dim_indexer_iterables is a list (one per dimension)
+                                # of lists of IntDimIndexer or SliceDimIndexer objects.
+                                dim_indexer_iterables <- lapply(self$dim_indexers, function(di) di$iter())
+                                dim_indexer_product <- get_list_product(dim_indexer_iterables)
+                                
+                                
+                                for(row_i in seq_len(length(dim_indexer_product))) {
+                                  dim_proj <- dim_indexer_product[[row_i]]
+                                  
+                                  chunk_coords <- list()
+                                  chunk_sel <- list()
+                                  out_sel <- list()
+                                  
+                                  if(!is.list(dim_proj)) {
+                                    dim_proj <- list(dim_proj)
+                                  }
+                                  
+                                  for(p in dim_proj) {
+                                    chunk_coords <- append(chunk_coords, p$dim_chunk_index)
+                                    # chunk_sel <- append(chunk_sel, p$dim_chunk_sel)
+                                    chunk_sel <- append(chunk_sel, list(p$dim_chunk_sel))
+                                    if(!is_na(p$dim_out_sel)) {
+                                      # out_sel <- append(out_sel, p$dim_out_sel)
+                                      out_sel <- append(out_sel, list(p$dim_out_sel))
+                                    }
+                                  }
+                                  
+                                  result <- append(result, ChunkProjection$new(
+                                    chunk_coords,
+                                    chunk_sel,
+                                    out_sel
+                                  ))
+                                }
+                                
+                                return(result)
+                              }
+                            )
+)
+
+# Reference: https://github.com/zarr-developers/zarr-python/blob/5dd4a0e6cdc04c6413e14f57f61d389972ea937c/zarr/indexing.py#L424
+
+#' The Order class.
+#' @title Order Class
+#' @docType class
+#' @description
+#'  TODO
+#' @rdname Order
+#' @keywords internal
+Order <- R6::R6Class("Order",
```

Comment:
Similarly, would you be able to add a test for this `Order` class?

Reply:
https://github.com/keller-mark/pizzarr/commit/adbdc6392f01c45881c10c2ecc983789c27e1822 let me know if this suffice